### DISCUSSION TITLE
gazebo simulator warning with TF

### DISCUSSION THREAD

**Discussion by jjnak** [HumanSupportRobot/QandA#77] [Q&A] [answered]
When running HSR gazebo simulator by
roslaunch hsrb_gazebo_launch hsrb_megaweb2015_world.launch

I noticed that the following warning shows up repeatedly, presumably with respect to TF lookup.
[ WARN] [1693383966.315755607, 6.047000000]: Could not get base to fixed frame transform, Lookup would require extrapolation 0.941000000s into the future.  Requested time 5.917000000 but the latest data is at time 4.976000000, when looking up transform from frame [base_footprint] to frame [odom]

I'd like to know if this shouldn't be a critical issue with it. Your help would be greatly appreciated.

-Jun

**Comment by yutaro-ishida** [comment]
Hi Jun,

The warning occur depending on the version of `laser_scan_matcher`.
Could you tell me the version using following command?

```
$ dpkg -l | grep laser-scan-matcher
```

Also, could you tell me the frequency of `/hsrb/base_pose` topic?

Yuta

**Reply by jjnak** [reply]
Thank you for your reply.

The version of `laser-scan-matcher` is 0.4.0-1focal.20230620.203330.

The frequency of `/hsrb/base_pose` is 
average rate: 10.000
	min: 0.100s max: 0.100s std dev: 0.00000s window: 9
by `rostopic hz /hsrb/base_pose`.

-Jun

**Reply by yutaro-ishida** [reply, marked as answer]
Hi Jun,

Thanks. In an environment without occuring warnings, the version is 0.3.3 and rate of `/hsrb/base_pose` is around 30 Hz.

Could you downgrade `laser-scan-mathcher` and related packages using following two commands?

1. Describe the package version specification in `/etc/apt/preferences`.
```
$ sudo sh -c '/bin/echo -e "Package: ros-noetic-laser-ortho-projector\nPin: version 0.3.3*\nPin-Priority: 1001\n\nPackage: ros-noetic-laser-scan-matcher\nPin: version 0.3.3*\nPin-Priority: 1001\n\nPackage: ros-noetic-laser-scan-sparsifier\nPin: version 0.3.3*\nPin-Priority: 1001\n\nPackage: ros-noetic-laser-scan-splitter\nPin: version 0.3.3*\nPin-Priority: 1001\n\nPackage: ros-noetic-ncd-parser\nPin: version 0.3.3*\nPin-Priority: 1001\n\nPackage: ros-noetic-polar-scan-matcher\nPin: version 0.3.3*\nPin-Priority: 1001\n\nPackage: ros-noetic-scan-to-cloud-converter\nPin: version 0.3.3*\nPin-Priority: 1001\n\nPackage: ros-noetic-scan-tools\nPin: version 0.3.3*\nPin-Priority: 1001" > /etc/apt/preferences'
```

2. Downgrade.
```
$ sudo apt-get insatll ros-noetic-laser-ortho-projector ros-noetic-laser-scan-matcher ros-noetic-laser-scan-sparsifier ros-noetic-laser-scan-splitter ros-noetic-ncd-parser ros-noetic-polar-scan-matcher ros-noetic-scan-to-cloud-converter ros-noetic-scan-tools
```

You might need `$ sudo apt update` before downgrade.

Yuta

**Reply by jjnak** [reply]
Thank you for your response. I think these instructions are described in the manual of software update for 23.04. I would like to confirm on which platform I should run these commands (item 1 and 2) in your reply either on the client PC or on HSR (or both).

Acccoding to the manual, 1 should be run on the client PC and 2 should be run on HSR. Your clarification would be greatly appreciated.

-Jun

**Reply by yutaro-ishida** [reply]
Hi Jun,

You should run both commands on your client PC.
`2. Downgrade` is alternative command of `update & upgrade` commands on [manual](https://docs.hsr.io/hsr_update_manual_en/23.04/chapters/update_pc.html).
(The difference is whether to downgrade only `laser-scan-mathcer` related packages, which is the cause of this issue, or to update all packages.)

Yuta

**Reply by jjnak** [reply]
Thank you for the instructions. I tried them on the client PC and it worked well.

-Jun
